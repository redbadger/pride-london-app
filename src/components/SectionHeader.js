// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import ContentPadding from "./ContentPadding";
import Text from "./Text";
import NumberBadge from "./NumberBadge";
import {
  sectionHeaderShadow,
  sectionHeaderBgColor,
  sectionHeaderTextColor
} from "../constants/colors";

type Props = { title: string, hasShadow?: boolean, badgeValue?: ?number };

const SectionHeader = ({ title, hasShadow, badgeValue }: Props) => {
  const containerStyles = hasShadow
    ? [styles.sectionHeader, styles.shadow]
    : styles.sectionHeader;
  const showBadge = badgeValue != null;
  return (
    <ContentPadding style={containerStyles}>
      <Text
        type="h2"
        style={styles.sectionHeaderText}
        accessible
        accessibilityTraits={["header"]}
      >
        {title}
      </Text>
      {showBadge ? (
        <View style={styles.badgeWrapper}>
          <NumberBadge value={((badgeValue: any): number)} />
        </View>
      ) : (
        <View />
      )}
    </ContentPadding>
  );
};
SectionHeader.defaultProps = {
  hasShadow: true,
  badgeValue: undefined
};

const styles = StyleSheet.create({
  sectionHeader: {
    height: 40,
    alignItems: "center",
    backgroundColor: sectionHeaderBgColor,
    flexDirection: "row"
  },
  shadow: {
    marginBottom: 6,
    // The below properties are required for ioS shadow
    shadowColor: sectionHeaderShadow,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 1,
    shadowRadius: 3,
    // The below properties are required for android shadow
    borderWidth: 0,
    elevation: 3
  },
  sectionHeaderText: {
    color: sectionHeaderTextColor
  },
  badgeWrapper: {
    marginLeft: 6
  }
});

export default SectionHeader;
