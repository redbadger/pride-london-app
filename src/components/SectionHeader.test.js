import React from "react";
import { shallow } from "enzyme";
import SectionHeader from "./SectionHeader";

it("renders correctly", () => {
  const output = shallow(<SectionHeader title="Section Header" />);
  expect(output).toMatchSnapshot();
});

it("renders correctly without shadow", () => {
  const output = shallow(
    <SectionHeader title="Section Header" hasShadow={false} />
  );
  expect(output).toMatchSnapshot();
});

it("renders correctly with badge text", () => {
  const output = shallow(
    <SectionHeader title="Section Header" badgeValue={2} />
  );
  expect(output).toMatchSnapshot();
});
