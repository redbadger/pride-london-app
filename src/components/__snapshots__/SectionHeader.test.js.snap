// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ContentPadding
  padding={Object {}}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f3f3f3",
        "flexDirection": "row",
        "height": 40,
      },
      Object {
        "borderWidth": 0,
        "elevation": 3,
        "marginBottom": 6,
        "shadowColor": "rgba(0, 0, 0, 0.14)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Text
    accessibilityTraits={
      Array [
        "header",
      ]
    }
    accessible={true}
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Object {
        "color": "rgb(45, 47, 127)",
      }
    }
    type="h2"
  >
    Section Header
  </Text>
  <Component />
</ContentPadding>
`;

exports[`renders correctly with badge text 1`] = `
<ContentPadding
  padding={Object {}}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f3f3f3",
        "flexDirection": "row",
        "height": 40,
      },
      Object {
        "borderWidth": 0,
        "elevation": 3,
        "marginBottom": 6,
        "shadowColor": "rgba(0, 0, 0, 0.14)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Text
    accessibilityTraits={
      Array [
        "header",
      ]
    }
    accessible={true}
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Object {
        "color": "rgb(45, 47, 127)",
      }
    }
    type="h2"
  >
    Section Header
  </Text>
  <Component
    style={
      Object {
        "marginLeft": 6,
      }
    }
  >
    <NumberBadge
      value={2}
    />
  </Component>
</ContentPadding>
`;

exports[`renders correctly without shadow 1`] = `
<ContentPadding
  padding={Object {}}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f3f3f3",
      "flexDirection": "row",
      "height": 40,
    }
  }
>
  <Text
    accessibilityTraits={
      Array [
        "header",
      ]
    }
    accessible={true}
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Object {
        "color": "rgb(45, 47, 127)",
      }
    }
    type="h2"
  >
    Section Header
  </Text>
  <Component />
</ContentPadding>
`;
